torch/distributed/device_mesh.py

```diff
@@ -287,10 +287,16 @@ def _init_process_groups(self):
                     # for each dim and append the groups
                     for dim_mesh in pg_ranks_by_dim:
                         subgroup_ranks = dim_mesh.tolist()
-                        # call new_group regardless of the current rank in the
-                        # pg or not, it's required that all ranks participate
-                        # in subgroup construction
-                        dim_group = new_group(ranks=subgroup_ranks)
+                        # if dim_group exists for given subgroup_ranks, we re-use it.
```

Comment:
actually given that we enable this, can we also delete the `_init_process_group` flag in the `DeviceMesh` constructor? we don't need that anymore